###### imports

In [75]:
import pandas as pd # for dataframe analysis
import numpy as np # for arrays functionality, i.e. .where()...
import re
from functools import partialmethod # for changing default of pandas .head()

###### defaults

[Set default .head() to 3 rows](https://medium.com/dunder-data/pandas-trick-1-change-the-default-number-of-rows-returned-from-the-head-method-bc7c21ce0d53)

In [76]:
# set default .head() to 3 rows per https://medium.com/dunder-data/
# pandas-trick-1-change-the-default-number-of-rows-returned-from-the-head-
# method-bc7c21ce0d53
pd.DataFrame.head = partialmethod(pd.DataFrame.head, n=3)

###### data cleaning

In [77]:
whfoods = pd.read_csv('./Data/whfoods.csv')

In [78]:
np.where(whfoods.columns.values=='Cheese, grass fed, cheddar, whole milk')[0][0]

246

In [79]:
whfoods.iloc[:,np.where(whfoods.columns.values=='Cheese, grass fed, cheddar, whole milk')[0][0]:np.where(whfoods.columns.values=='Cheese, grass fed, cheddar, whole milk')[0][0]+2
            ].head(n = 5)

Cheese, grass fed, cheddar, whole milk Unnamed: 247
0      BASIC MACRONUTRIENTS AND CALORIES          NaN
1                               nutrient       amount
2                                    NaN          NaN
3                                Protein       7.06 g
4                          Carbohydrates       0.36 g

In [80]:
# 

In [81]:
# remove extraneous rows, columns.

whfoods = whfoods.drop([0,175,176]).dropna(how='all',axis=1).dropna(how='all',
                                                                    axis=0)

In [82]:
whfoods.index = range(whfoods.shape[0])

In [83]:
whfoods

Asparagus, Cooked Unnamed: 1 Unnamed: 2  \
0                         nutrient     amount     DRI/DV   
1                              NaN        NaN        (%)   
2                          Protein     4.32 g          9   
3                    Carbohydrates     7.40 g          3   
4                      Fat - total     0.40 g          1   
..                             ...        ...        ...   
168  Artificial Sweeteners (Total)     #NAME?        NaN   
169                      Aspartame     #NAME?        NaN   
170                      Saccharin     #NAME?        NaN   
171                        Alcohol     0.00 g        NaN   
172                       Caffeine    0.00 mg        NaN   

               Avocado, cubed, raw Unnamed: 4 Unnamed: 5  \
0                         nutrient     amount     DRI/DV   
1                              NaN        NaN        (%)   
2                          Protein     3.00 g          6   
3                    Carbohydrates    12.80 g          6   
4                      Fat - total    21.99 g         28   
..                             ...        ...        ...   
168  Artificial Sweeteners (Total)     #NAME?        NaN   
169                      Aspartame     #NAME?        NaN   
170                      Saccharin     #NAME?        NaN   
171                        Alcohol     0.00 g        NaN   
172                       Caffeine    0.00 mg        NaN   

               Beet Greens, boiled Unnamed: 7 Unnamed: 8  \
0                         nutrient     amount     DRI/DV   
1                              NaN        NaN        (%)   
2                          Protein     3.70 g          7   
3                    Carbohydrates     7.86 g          3   
4                      Fat - total     0.29 g          0   
..                             ...        ...        ...   
168  Artificial Sweeteners (Total)     #NAME?        NaN   
169                      Aspartame     #NAME?        NaN   
170                      Saccharin     #NAME?        NaN   
171                        Alcohol     0.00 g        NaN   
172                       Caffeine    0.00 mg        NaN   

             Beets, sliced, cooked  ... Unnamed: 341  \
0                         nutrient  ...       DRI/DV   
1                              NaN  ...          (%)   
2                          Protein  ...            0   
3                    Carbohydrates  ...            0   
4                      Fat - total  ...            0   
..                             ...  ...          ...   
168  Artificial Sweeteners (Total)  ...          NaN   
169                      Aspartame  ...          NaN   
170                      Saccharin  ...          NaN   
171                        Alcohol  ...          NaN   
172                       Caffeine  ...          NaN   

                       Sage, dried Unnamed: 343 Unnamed: 344  \
0                         nutrient       amount       DRI/DV   
1                              NaN          NaN          (%)   
2                          Protein       0.15 g            0   
3                    Carbohydrates       0.85 g            0   
4                      Fat - total       0.18 g            0   
..                             ...          ...          ...   
168  Artificial Sweeteners (Total)       #NAME?          NaN   
169                      Aspartame       #NAME?          NaN   
170                      Saccharin       #NAME?          NaN   
171                        Alcohol       0.00 g          NaN   
172                       Caffeine      0.00 mg          NaN   

                      Thyme, fresh Unnamed: 346 Unnamed: 347  \
0                         nutrient       amount       DRI/DV   
1                              NaN          NaN          (%)   
2                          Protein       0.27 g            1   
3                    Carbohydrates       1.17 g            1   
4                      Fat - total       0.08 g            0   
..                          

In [84]:
# investigating dimensions of dataset

nfoods = len(whfoods.columns)/3 # there are 117 foods in the dataset.
nfoods
nnutrients = whfoods.shape[0]
whfoods.shape # There are 

(173, 351)

In [85]:
whfoods

Asparagus, Cooked Unnamed: 1 Unnamed: 2  \
0                         nutrient     amount     DRI/DV   
1                              NaN        NaN        (%)   
2                          Protein     4.32 g          9   
3                    Carbohydrates     7.40 g          3   
4                      Fat - total     0.40 g          1   
..                             ...        ...        ...   
168  Artificial Sweeteners (Total)     #NAME?        NaN   
169                      Aspartame     #NAME?        NaN   
170                      Saccharin     #NAME?        NaN   
171                        Alcohol     0.00 g        NaN   
172                       Caffeine    0.00 mg        NaN   

               Avocado, cubed, raw Unnamed: 4 Unnamed: 5  \
0                         nutrient     amount     DRI/DV   
1                              NaN        NaN        (%)   
2                          Protein     3.00 g          6   
3                    Carbohydrates    12.80 g          6   
4                      Fat - total    21.99 g         28   
..                             ...        ...        ...   
168  Artificial Sweeteners (Total)     #NAME?        NaN   
169                      Aspartame     #NAME?        NaN   
170                      Saccharin     #NAME?        NaN   
171                        Alcohol     0.00 g        NaN   
172                       Caffeine    0.00 mg        NaN   

               Beet Greens, boiled Unnamed: 7 Unnamed: 8  \
0                         nutrient     amount     DRI/DV   
1                              NaN        NaN        (%)   
2                          Protein     3.70 g          7   
3                    Carbohydrates     7.86 g          3   
4                      Fat - total     0.29 g          0   
..                             ...        ...        ...   
168  Artificial Sweeteners (Total)     #NAME?        NaN   
169                      Aspartame     #NAME?        NaN   
170                      Saccharin     #NAME?        NaN   
171                        Alcohol     0.00 g        NaN   
172                       Caffeine    0.00 mg        NaN   

             Beets, sliced, cooked  ... Unnamed: 341  \
0                         nutrient  ...       DRI/DV   
1                              NaN  ...          (%)   
2                          Protein  ...            0   
3                    Carbohydrates  ...            0   
4                      Fat - total  ...            0   
..                             ...  ...          ...   
168  Artificial Sweeteners (Total)  ...          NaN   
169                      Aspartame  ...          NaN   
170                      Saccharin  ...          NaN   
171                        Alcohol  ...          NaN   
172                       Caffeine  ...          NaN   

                       Sage, dried Unnamed: 343 Unnamed: 344  \
0                         nutrient       amount       DRI/DV   
1                              NaN          NaN          (%)   
2                          Protein       0.15 g            0   
3                    Carbohydrates       0.85 g            0   
4                      Fat - total       0.18 g            0   
..                             ...          ...          ...   
168  Artificial Sweeteners (Total)       #NAME?          NaN   
169                      Aspartame       #NAME?          NaN   
170                      Saccharin       #NAME?          NaN   
171                        Alcohol       0.00 g          NaN   
172                       Caffeine      0.00 mg          NaN   

                      Thyme, fresh Unnamed: 346 Unnamed: 347  \
0                         nutrient       amount       DRI/DV   
1                              NaN          NaN          (%)   
2                          Protein       0.27 g            1   
3                    Carbohydrates       1.17 g            1   
4                      Fat - total       0.08 g            0   
..                          

In [86]:
whfoods.iloc[:,0].values

array(['nutrient', nan, 'Protein', 'Carbohydrates', 'Fat - total',
       'Dietary Fiber', 'Calories', 'MACRONUTRIENT AND CALORIE DETAIL',
       'nutrient', nan, 'Carbohydrate:', 'Starch', 'Total Sugars',
       'Monosaccharides', 'Fructose', 'Glucose', 'Galactose',
       'Disaccharides', 'Lactose', 'Maltose', 'Sucrose', 'Soluble Fiber',
       'Insoluble Fiber', 'Other Carbohydrates', 'Fat:',
       'Monounsaturated Fat', 'Polyunsaturated Fat', 'Saturated Fat',
       'Trans Fat', 'Calories from Fat', 'Calories from Saturated Fat',
       'Calories from Trans Fat', 'Cholesterol', 'Water',
       'MICRONUTRIENTS', 'nutrient', nan, 'Vitamins',
       'Water-Soluble Vitamins', 'B-Complex Vitamins', 'Vitamin B1',
       'Vitamin B2', 'Vitamin B3', 'Vitamin B3 (Niacin Equivalents)',
       'Vitamin B6', 'Vitamin B12', 'Biotin', 'Choline', 'Folate',
       'Folate (DFE)', 'Folate (food)', 'Pantothenic Acid', 'Vitamin C',
       'Fat-Soluble Vitamins', 'Vitamin A (Retinoids and Carotenoids

There are under 173 nutrient categories since some of the rows correspond to supercategories such as 'Minerals','INDIVIDUAL FATTY ACIDS', 'Monounsaturated Fats', 'INDIVIDUAL AMINO ACIDS', 'OTHER COMPONENTS', etc. as well as corresponding extraneous rows such as: 'nutrient', nan.

The dataset could be simplified by removing rows with all 0.00 g, mg etc or all -- mgsuch as with the sweeteners, caffeine, alcohol.

I'll create two dictionaries, one of the foods and the raw nutrient values and another of the foods and DRI/DV.  These can be used to generate data frames.  The raw nutrient values and the DRI/DV info can be used to generate the 100% DRI/DV values for each nutrient.  Rounding errors can be overcome by using the food with the highest DRI/DV for a given nutrient to generate the recommendation.

###### column name formatting

In [87]:
# gather all the foods into a list
foods = [food for inx,food in enumerate(whfoods.columns) if inx%3 ==0]
len(foods) # verify the number of foods

117

In [88]:
# 
new_cols = []
for i, food in enumerate(foods):
    # Replace spaces with a single underscore
    food = re.sub(r' +', '_', food)
    # Replace commas with a single underscore
    food = re.sub(r',+', '_', food)
    # Remove duplicate underscores
    food = re.sub(r'_+', '_', food)
    new_cols = new_cols + [food, f'NV_{food}', f'DRVs_{food}']

In [89]:
whfoods.columns = new_cols

In [90]:
whfoods.head()

Asparagus_Cooked NV_Asparagus_Cooked DRVs_Asparagus_Cooked  \
0         nutrient              amount                DRI/DV   
1              NaN                 NaN                   (%)   
2          Protein              4.32 g                     9   

  Avocado_cubed_raw NV_Avocado_cubed_raw DRVs_Avocado_cubed_raw  \
0          nutrient               amount                 DRI/DV   
1               NaN                  NaN                    (%)   
2           Protein               3.00 g                      6   

  Beet_Greens_boiled NV_Beet_Greens_boiled DRVs_Beet_Greens_boiled  \
0           nutrient                amount                  DRI/DV   
1                NaN                   NaN                     (%)   
2            Protein                3.70 g                       7   

  Beets_sliced_cooked  ... DRVs_Rosemary_fresh Sage_dried NV_Sage_dried  \
0            nutrient  ...              DRI/DV   nutrient        amount   
1                 NaN  ...                 (%)        NaN           NaN   
2             Protein  ...                   0    Protein        0.15 g   

  DRVs_Sage_dried Thyme_fresh NV_Thyme_fresh DRVs_Thyme_fresh Turmeric_ground  \
0          DRI/DV    nutrient         amount           DRI/DV        nutrient   
1             (%)         NaN            NaN              (%)             NaN   
2               0     Protein         0.27 g                1         Protein   

  NV_Turmeric_ground DRVs_Turmeric_ground  
0             amount               DRI/DV  
1                NaN                  (%)  
2             0.34 g                    1  

[3 rows x 351 columns]

In [93]:
# gather reformatted food names into a list
foods = [food for inx,food in enumerate(whfoods.columns) if inx%3 == 0]
len(foods) # verify the number of foods

117

In [94]:
nnutrients

173

###### separate nutrient val, drv info

In [111]:
# collect a nested dictionary of... {food: {nutrient:nutrient_val}}
nutrient_vals = \
{food:
    {whfoods.loc[i,food]: 
     whfoods.iloc[i,int(np.where(whfoods.columns.values==food)[0][0])+1] 
     for i in range (2, nnutrients)
    } for food in foods
}

# collect a nested dictionary of... {food: {nutrient:nutrient_drv}}
nutrient_drv = \
{food:
    {whfoods.loc[i,food]: 
     whfoods.iloc[i,int(np.where(whfoods.columns.values==food)[0][0])+2] 
     for i in range (2, nnutrients)
    } for food in foods
}

# convert to DatFrame
nutrient_vals_df = pd.DataFrame(nutrient_vals)
nutrient_drv_df = pd.DataFrame(nutrient_drv)

# Transposing so that the foods are like "observations" in long format
nutrient_vals_df = nutrient_vals_df.T
nutrient_drv_df = nutrient_drv_df.T

In [118]:
nutrient_vals_df.columns.values

array(['Protein', 'Carbohydrates', 'Fat - total', 'Dietary Fiber',
       'Calories', 'MACRONUTRIENT AND CALORIE DETAIL', 'nutrient', nan,
       'Carbohydrate:', 'Starch', 'Total Sugars', 'Monosaccharides',
       'Fructose', 'Glucose', 'Galactose', 'Disaccharides', 'Lactose',
       'Maltose', 'Sucrose', 'Soluble Fiber', 'Insoluble Fiber',
       'Other Carbohydrates', 'Fat:', 'Monounsaturated Fat',
       'Polyunsaturated Fat', 'Saturated Fat', 'Trans Fat',
       'Calories from Fat', 'Calories from Saturated Fat',
       'Calories from Trans Fat', 'Cholesterol', 'Water',
       'MICRONUTRIENTS', 'Vitamins', 'Water-Soluble Vitamins',
       'B-Complex Vitamins', 'Vitamin B1', 'Vitamin B2', 'Vitamin B3',
       'Vitamin B3 (Niacin Equivalents)', 'Vitamin B6', 'Vitamin B12',
       'Biotin', 'Choline', 'Folate', 'Folate (DFE)', 'Folate (food)',
       'Pantothenic Acid', 'Vitamin C', 'Fat-Soluble Vitamins',
       'Vitamin A (Retinoids and Carotenoids)',
       'Vitamin A Internationa

First, I need to gather units.  I can look for 
set #NAME? to 0.00

Idea: remove all columns that have uniform values

In [130]:
# credit: ChatGPT
def remove_columns_with_same_value(df):
    # Calculate the number of unique values in each column
    unique_counts = df.nunique()

    # Get the column names where the number of unique values is 1
    columns_to_remove = unique_counts[(unique_counts == 1)|(unique_counts == 0)].index

    # Remove the columns from the dataframe
    df = df.drop(columns=columns_to_remove)

    return df

In [119]:
nutrient_vals_df = \
nutrient_vals_df.drop(
    columns=['nutrient', 'MACRONUTRIENT AND CALORIE DETAIL',np.nan])

Protein Carbohydrates Fat - total Dietary Fiber  \
Asparagus_Cooked             4.32 g        7.40 g      0.40 g        3.60 g   
Avocado_cubed_raw            3.00 g       12.80 g     21.99 g       10.05 g   
Beet_Greens_boiled           3.70 g        7.86 g      0.29 g        4.18 g   
Beets_sliced_cooked          2.86 g       16.93 g      0.31 g        3.40 g   
Bell_Peppers_sliced_red_raw  0.91 g        5.55 g      0.28 g        1.85 g   
...                             ...           ...         ...           ...   
Peppermint_leaves_fresh      0.28 g        1.13 g      0.07 g        0.61 g   
Rosemary_fresh               0.14 g        0.87 g      0.25 g        0.59 g   
Sage_dried                   0.15 g        0.85 g      0.18 g        0.56 g   
Thyme_fresh                  0.27 g        1.17 g      0.08 g        0.67 g   
Turmeric_ground              0.34 g        2.86 g      0.43 g        0.93 g   

                            Calories Carbohydrate:  Starch Total Sugars  \
Asparagus_Cooked                39.6           NaN  #NAME?       2.34 g   
Avocado_cubed_raw                240           NaN  #NAME?       0.99 g   
Beet_Greens_boiled             38.88           NaN  #NAME?       0.86 g   
Beets_sliced_cooked             74.8           NaN  0.00 g      13.53 g   
Bell_Peppers_sliced_red_raw    28.52           NaN  0.00 g       3.70 g   
...                              ...           ...     ...          ...   
Peppermint_leaves_fresh         5.32           NaN  #NAME?       #NAME?   
Rosemary_fresh                   5.5           NaN  #NAME?       #NAME?   
Sage_dried                      4.41           NaN  #NAME?       0.02 g   
Thyme_fresh                     4.85           NaN  #NAME?       #NAME?   
Turmeric_ground                15.58           NaN  #NAME?       0.14 g   

                            Monosaccharides Fructose  ... Glycerol Inositol  \
Asparagus_Cooked                     2.18 g   1.42 g  ...   #NAME?   #NAME?   
Avocado_cubed_raw                    0.89 g   0.18 g  ...   #NAME?   #NAME?   
Beet_Greens_boiled                   #NAME?   #NAME?  ...   #NAME?   #NAME?   
Beets_sliced_cooked                  #NAME?   #NAME?  ...   0.00 g   0.00 g   
Bell_Peppers_sliced_red_raw          3.70 g   1.99 g  ...   0.00 g   0.00 g   
...                                     ...      ...  ...      ...      ...   
Peppermint_leaves_fresh              #NAME?   #NAME?  ...   #NAME?   #NAME?   
Rosemary_fresh                       #NAME?   #NAME?  ...   #NAME?   #NAME?   
Sage_dried                           #NAME?   #NAME?  ...   #NAME?   #NAME?   
Thyme_fresh                          #NAME?   #NAME?  ...   #NAME?   #NAME?   
Turmeric_ground                      0.04 g   0.02 g  ...   #NAME?   #NAME?   

                            Mannitol Sorbitol Xylitol  \
Asparagus_Cooked              #NAME?   #NAME?  #NAME?   
Avocado_cubed_raw             #NAME?   #NAME?  #NAME?   
Beet_Greens_boiled            #NAME?   #NAME?  #NAME?   
Beets_sliced_cooked           0.00 g   0.00 g  0.00 g   
Bell_Peppers_sliced_red_raw   0.00 g   0.00 g  0.00 g   
...                              ...      ...     ...   
Peppermint_leaves_fresh       #NAME?   #NAME?  #NAME?   
Rosemary_fresh                #NAME?   #NAME?  #NAME?   
Sage_dried                    #NAME?   #NAME?  #NAME?   
Thyme_fresh                   #NAME?   #NAME?  #NAME?   
Turmeric_ground               #NAME?   #NAME?  #NAME?   

                            Artificial Sweeteners (Total) Aspartame Saccharin  \
Asparagus_Cooked                                   #NAME?    #NAME?    #NAME?   
Avocado_cubed_raw                                  #NAME?    #NAME?    #NAME?   
Beet_Greens_boiled                                 #NAME?    #NAME?    #NAME?   
Beets_sliced_cooked                                #NAME?    #NAME?    #NAME?   
Bell_Peppers_sliced_red_raw                       0.00 mg   0.00 mg   0.00 mg   
...                                                   ... 

In [131]:
nutrient_vals_df = remove_columns_with_same_value(nutrient_vals_df)

In [132]:
nutrient_vals_df

Protein Carbohydrates Fat - total Dietary Fiber  \
Asparagus_Cooked             4.32 g        7.40 g      0.40 g        3.60 g   
Avocado_cubed_raw            3.00 g       12.80 g     21.99 g       10.05 g   
Beet_Greens_boiled           3.70 g        7.86 g      0.29 g        4.18 g   
Beets_sliced_cooked          2.86 g       16.93 g      0.31 g        3.40 g   
Bell_Peppers_sliced_red_raw  0.91 g        5.55 g      0.28 g        1.85 g   
...                             ...           ...         ...           ...   
Peppermint_leaves_fresh      0.28 g        1.13 g      0.07 g        0.61 g   
Rosemary_fresh               0.14 g        0.87 g      0.25 g        0.59 g   
Sage_dried                   0.15 g        0.85 g      0.18 g        0.56 g   
Thyme_fresh                  0.27 g        1.17 g      0.08 g        0.67 g   
Turmeric_ground              0.34 g        2.86 g      0.43 g        0.93 g   

                            Calories  Starch Total Sugars Monosaccharides  \
Asparagus_Cooked                39.6  #NAME?       2.34 g          2.18 g   
Avocado_cubed_raw                240  #NAME?       0.99 g          0.89 g   
Beet_Greens_boiled             38.88  #NAME?       0.86 g          #NAME?   
Beets_sliced_cooked             74.8  0.00 g      13.53 g          #NAME?   
Bell_Peppers_sliced_red_raw    28.52  0.00 g       3.70 g          3.70 g   
...                              ...     ...          ...             ...   
Peppermint_leaves_fresh         5.32  #NAME?       #NAME?          #NAME?   
Rosemary_fresh                   5.5  #NAME?       #NAME?          #NAME?   
Sage_dried                      4.41  #NAME?       0.02 g          #NAME?   
Thyme_fresh                     4.85  #NAME?       #NAME?          #NAME?   
Turmeric_ground                15.58  #NAME?       0.14 g          0.04 g   

                            Fructose Glucose  ... Sugar Alcohols (Total)  \
Asparagus_Cooked              1.42 g  0.76 g  ...                 #NAME?   
Avocado_cubed_raw             0.18 g  0.56 g  ...                 #NAME?   
Beet_Greens_boiled            #NAME?  #NAME?  ...                 #NAME?   
Beets_sliced_cooked           #NAME?  #NAME?  ...                 0.00 g   
Bell_Peppers_sliced_red_raw   1.99 g  1.71 g  ...                 0.00 g   
...                              ...     ...  ...                    ...   
Peppermint_leaves_fresh       #NAME?  #NAME?  ...                 #NAME?   
Rosemary_fresh                #NAME?  #NAME?  ...                 #NAME?   
Sage_dried                    #NAME?  #NAME?  ...                 #NAME?   
Thyme_fresh                   #NAME?  #NAME?  ...                 #NAME?   
Turmeric_ground               0.02 g  0.02 g  ...                 #NAME?   

                            Glycerol Inositol Mannitol Sorbitol Xylitol  \
Asparagus_Cooked              #NAME?   #NAME?   #NAME?   #NAME?  #NAME?   
Avocado_cubed_raw             #NAME?   #NAME?   #NAME?   #NAME?  #NAME?   
Beet_Greens_boiled            #NAME?   #NAME?   #NAME?   #NAME?  #NAME?   
Beets_sliced_cooked           0.00 g   0.00 g   0.00 g   0.00 g  0.00 g   
Bell_Peppers_sliced_red_raw   0.00 g   0.00 g   0.00 g   0.00 g  0.00 g   
...                              ...      ...      ...      ...     ...   
Peppermint_leaves_fresh       #NAME?   #NAME?   #NAME?   #NAME?  #NAME?   
Rosemary_fresh                #NAME?   #NAME?   #NAME?   #NAME?  #NAME?   
Sage_dried                    #NAME?   #NAME?   #NAME?   #NAME?  #NAME?   
Thyme_fresh                   #NAME?   #NAME?   #NAME?   #NAME?  #NAME?   
Turmeric_ground               #NAME?   #NAME?   #NAME?   #NAME?  #NAME?   

                            Artificial Sweeteners (Total) Aspartame Saccharin  \
Asparagus_Cooked                                   #NAME?    #NAME?    #NAME?   
Avocado_cubed_raw                                  #NAME?    #NAME?    #NAME?   
Beet_Greens_boiled                                 #NAME?    #NAME?    #NAME?   
Beets_sliced_coo

In [136]:
nutrient_vals_df['15:1 Pentadecenoic']

Asparagus_Cooked               #NAME?
Avocado_cubed_raw              0.00 g
Beet_Greens_boiled             0.00 g
Beets_sliced_cooked            0.00 g
Bell_Peppers_sliced_red_raw    0.00 g
                                ...  
Peppermint_leaves_fresh        #NAME?
Rosemary_fresh                 #NAME?
Sage_dried                     0.00 g
Thyme_fresh                    0.00 g
Turmeric_ground                0.00 g
Name: 15:1 Pentadecenoic, Length: 117, dtype: object

In [ ]:
nutrient_vals_df.query("Aspartame !=")

In [137]:
nutrient_vals_df.nunique().sort_values()[0:20]

Caffeine                          2
15:1 Pentadecenoic                2
Saccharin                         2
Glycerol                          2
Inositol                          2
Mannitol                          2
Sorbitol                          2
Artificial Sweeteners (Total)     2
Lactic Acid                       2
Aspartame                         2
14:1 Myristoleic                  3
Acetic Acid                       3
22:5 Docosapentaenoic (DPA)       3
Sugar Alcohols (Total)            3
Xylitol                           3
24:1 Nervonic                     4
20:3 Eicosatrienoic               4
15:0 Pentadecanoic                4
Malic Acid                        5
18:2 Conjugated Linoleic (CLA)    5
dtype: int64

In [114]:
conversions = 0
for nutrient in nutrient_drv_df.columns.values[[0,1]]:
    try: 
        nutrient_drv_df[[nutrient]] = nutrient_drv_df[[nutrient]].astype("int")
        conversions += 1
    except: pass

In [115]:
conversions

2

In [ ]:
nutrient_drv_df

In [ ]:
nutrient_drv_df.nunique()[10:40]

To simplify & clean the dataset, I'll collect the "nutrient_drv" columns that don't carry any information, i.e. have the same values for all the foods.

In [ ]:
[nutrient_drv_df[]for ]